source/Engine.cpp

```diff
@@ -634,6 +635,16 @@ void Engine::Step(bool isActive)
 			}
 		}
 
+	// Create missile overlays.
+	missileLabels.clear();
+	if(Preferences::Has("Show missile overlays"))
+		for(const Projectile &projectile : projectiles)
+		{
+			Point pos = projectile.Position() - center;
+			if((pos.Length() < max(Screen::Width(), Screen::Height()) * .5 / zoom)  && projectile.MissileStrength() && projectile.GetGovernment()->IsEnemy())
```

Comment:
Take advantage of short-circuiting and put the most expensive calculation last.
```suggestion
			if(projectile.MissileStrength() && projectile.GetGovernment()->IsEnemy() && (pos.Length() < max(Screen::Width(), Screen::Height()) * .5 / zoom))
```